osu.Game.Rulesets.Mania/Scoring/ManiaHealthProcessor.cs

```diff
@@ -15,6 +15,14 @@ public ManiaHealthProcessor(double drainStartTime)
         {
         }
 
+        protected override double ComputeDrainRate()
+        {
+            // Base call is run only to compute HP recovery.
```

Comment:
I'm not entirely sure how to feel about this. From what I recall you saying, this replicates stable terminal breakage to the T (if I remember correctly, for mania stable also computes HP recovery assuming passive drain is active but then... doesn't drain), but it's still rather unsightly...

If I were to suggest anything I'd probably just expand on the comment more:

```suggestion
            // Base call is run only to compute HP recovery (namely, `HpMultiplierNormal`).
            // This closely mirrors (broken) behaviour of stable and as such is preserved unchanged.
```

I'm not sure it's feasible or worth it to refactor this further in a way that doesn't require the "fake" base call.